### ISSUE TITLE
Texture need support gpu buffer data in common API

### ISSUE THREAD

**Issue by xiaowei-guan** [flutter/flutter#83693]
## Use case
As a developer of other platforms, we use the common API(shell/platform/common) to develop plugins . In the common API, I found that txture only supports pixel buffer data.As we know, there are performance problems when using pixel buffer data because it copies data from CPU to GPU. 

## Proposal
 When using Texture, we need Common API to support GPU buffer Data:
 1.we can add a new struct for gpu buffer data.
 ```cpp// An image buffer object.
typedef struct {
  // The gpu data buffer.
  const void* buffer;
  // Width of the gpu buffer.
  size_t width;
  // Height of the gpu buffer.
  size_t height;
} FlutterDesktopGpuBuffer;
```
2.we can add a new callback to obtain gpu buffer data pointer when tiggle gl_external_texture_frame_callback.
3.Add a destruction callback when GPU render finished.

**Comment by xiaowei-guan** [comment]
#https://github.com/flutter-tizen/engine/pull/86

**Comment by xiaowei-guan** [comment]
#flutter-tizen/engine/pull/86

**Comment by HidenoriMatsubayashi** [comment]
@xiaowei-guan Great work. Great work. I wanted this function.

**Comment by lucaszanella** [comment]
@xiaowei-guan could you give me some context on this? I've seen other PRs about external texture support for desktop. Does external texture work on desktop yet? I remember I was able to use it with one of the old PRs but it got rejected because it was based on GLFW.

Is it possible to register a texture on desktop and render to it already?

**Comment by swift-kim** [comment]
@lucaszanella It is already possible to register external textures on Windows (https://github.com/flutter/engine/pull/19405). This proposal adds a new texture type support on top of the current pixel buffer-based texture support because the pixel buffer textures may lead to a performance issue on non-desktop low-end devices.

**Comment by zanderso** [comment]
Unfortunately, we probably can't support this in a generic way for all OpenGL users from the embedding API. This may need to be handled by embedders in an embedder specific way. /cc @cbracken

**Comment by zanderso** [comment]
Also /cc @arbreng

**Comment by swift-kim** [comment]
@xiaowei-guan This issue might be closed as well in favor of https://github.com/flutter/engine/pull/26840.

**Comment by xiaowei-guan** [comment]
> @xiaowei-guan This issue might be closed as well in favor of [flutter/engine#26840](https://github.com/flutter/engine/pull/26840).

Ok

**Comment by xiaowei-guan** [comment]
This PR supports GPU buffer texture : https://github.com/flutter/engine/pull/26840
